### ISSUE TITLE
Big Sur support?

### ISSUE THREAD

**Issue by ethan-leba** [godbout/kindaVim.theapp#16]
Hi, this project looks really cool! Unfortunately this app doesn't support Big Sur -- would it be possible to support it? (or does it use some Monterey specific functionality?)

**Comment by godbout** [comment]
hello Sir!

short answer: no, sorry.
a bit longer answer: started with Catalina, then Big Sur, but ultimately needed Swift 5.5 and SwiftUI 3, so Monterey only. i'm kinda sad also. iMac 27 stuck on Catalina here.

**Comment by godbout** [comment]
ok to be fair *needed* is not the right word. you could make kV work with AppKit. but i have no knowledge of AppKit or i had no intention to learn to due to pure laziness. and the fact that SwiftUI is the way forward.

ok this answer seems more accurate.

**Comment by ethan-leba** [comment]
Fair enough, i'll have to revisit when I get around to the upgrade then :)

**Comment by godbout** [comment]
hopefully being able to use kindaVim will make the upgrade worth it.


![](https://www.icegif.com/wp-content/uploads/uncomfortable-white-people.gif)

**Comment by clouedoc** [comment]
> hopefully being able to use kindaVim will make the upgrade worth it.
> 
> ![](https://camo.githubusercontent.com/ca279742cc70bc55484618cb2ecef1f799545b2c64406948d60c4f6a0c0d85bc/68747470733a2f2f7777772e6963656769662e636f6d2f77702d636f6e74656e742f75706c6f6164732f756e636f6d666f727461626c652d77686974652d70656f706c652e676966) [ ](https://camo.githubusercontent.com/ca279742cc70bc55484618cb2ecef1f799545b2c64406948d60c4f6a0c0d85bc/68747470733a2f2f7777772e6963656769662e636f6d2f77702d636f6e74656e742f75706c6f6164732f756e636f6d666f727461626c652d77686974652d70656f706c652e676966) !


I'm currently considering upgrading my mac because of kindaVim. 
But I'm kinda afraid that all my apps are going to crash when upgrading.
But I kinda need `jk` because I want to look cooler.

![image](https://user-images.githubusercontent.com/13921610/174498777-9b97ea83-c9f7-4ce6-b378-26759bcac8ec.png)

## Checklist

- [ ] Music things
- [ ] Programming things

**Comment by godbout** [comment]
> > hopefully being able to use kindaVim will make the upgrade worth it.
> > ![](https://camo.githubusercontent.com/ca279742cc70bc55484618cb2ecef1f799545b2c64406948d60c4f6a0c0d85bc/68747470733a2f2f7777772e6963656769662e636f6d2f77702d636f6e74656e742f75706c6f6164732f756e636f6d666f727461626c652d77686974652d70656f706c652e676966) [ ](https://camo.githubusercontent.com/ca279742cc70bc55484618cb2ecef1f799545b2c64406948d60c4f6a0c0d85bc/68747470733a2f2f7777772e6963656769662e636f6d2f77702d636f6e74656e742f75706c6f6164732f756e636f6d666f727461626c652d77686974652d70656f706c652e676966) !
> 
> I'm currently considering upgrading my mac because of kindaVim. But I'm kinda afraid that all my apps are going to crash when upgrading. But I kinda need `jk` because I want to look cooler.
> 
> ![image](https://user-images.githubusercontent.com/13921610/174498777-9b97ea83-c9f7-4ce6-b378-26759bcac8ec.png)
> 
> ## Checklist
> * [ ]  Music things
> * [ ]  Programming things

funny, i actually had another look at supporting Big Sur last week. the gist of the conclusion is:
1. the whole Preferences would need to be rewritten (or at least done again once, bug with Big Sur APIs)
2. if it was the backend that would be an issue, it would be easier to test. because it's the Preferences, i would need access to a Big Sur machine to test (which is not the case)
3. with Ventura coming soon, makes even less sense to support backward rather than forward. there's even stuff in Ventura that will/should help, but i'm not gonna drop Monterey, so it's gonna be painful

so the decision was the same as before, no Big Sur.

in your case what's the specific issues? old laptop? old apps that are not getting updated anymore? the switch from Catalina to Big Sur was bigger. Monterey is just a continuity of Big Sur. so i would expect the update to go smoothly?

**Comment by clouedoc** [comment]
Yeah, I definitely think that you should support only the latest major releases because kV kinda looks like a pain to maintain.

I have an M1, but I had bad experiences upgrading in the past because all of the legacy software I use for my music projects. Each time, I need to reauthenticate (= pirate 🌝) all of them. And sometimes, some amazing virtual instrument that I was using won't work because of dark issue X on version Y.

TL;DR I should have a computer for music and a computer for programming. Don't worry about it 😛

**Comment by godbout** [comment]
ah, i feel the pain 😅️
i use to carry a MBA ten years ago when i was doing music. MBA plugged in some box, connected to guitar pedals, and keyboard. then on Monday the MBA would be back for coding. a nightmare 😂️
